core/shared/src/main/scala/fs2/concurrent/Queue.scala

```diff
@@ -465,6 +496,24 @@ object InspectableQueue {
     def bounded[F[_], A](maxSize: Int)(implicit F: Concurrent[F]): G[InspectableQueue[F, A]] =
       forStrategy(Queue.Strategy.boundedFifo[A](maxSize))(_.headOption)(_.size)
 
+    /** Creates a bounded queue terminated by enqueueing `None`. All elements before `None` are preserved. */
+    def boundedNoneTerminated[F[_], A](
+        maxSize: Int
+    )(implicit F: Concurrent[F]): G[NoneTerminatedInspectableQueue[F, A]] =
+      forStrategyNoneTerminated[F, (Boolean, SizedQueue[A]), A](
+        PubSub.Strategy.closeDrainFirst(Queue.Strategy.boundedFifo(maxSize))
+      ) { case (_, h) => h.headOption } { case (_, h) => h.size }
+
+    /** Created an unbounded queue terminated by enqueueing `None`. All elements before `None`. */
+    def noneTerminated[F[_], A](implicit
+        F: Concurrent[F]
+    ): G[NoneTerminatedInspectableQueue[F, A]] =
```

Comment:
```suggestion
    def noneTerminated[F[_]: Concurrent, A]: G[NoneTerminatedInspectableQueue[F, A]] =
```

No explicit use of the `F` parameter.